src/components/Provider.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+
+import { IProviderContext, IStore, ISubscriber } from "../types";
+
+const initialContext = {
+  store: {},
+  update: () => {},
+  dispatch: () => {}
+};
+
+const NucleoContext = React.createContext<IProviderContext>(initialContext);
+
+interface IProps {
+  store: IStore,
+}
+
+interface IState {
+  store: Object,
+}
+
+class Provider extends React.Component<IProps, IState> {
+  constructor(props: IProps) {
+    super(props);
+
+    if (!props.store) throw new Error("Store props is needed");
+
+    this.state = {
+      store: props.store.getStore() || {}
+    };
+  }
+
+  componentDidMount() {
+    const { store } = this.props;
+    const { subscribe } = store;
+    subscribe(this.contractsListener.bind(this));
+  }
+
+  contractsListener({ contractName }: ISubscriber) {
```

Comment:
use `arrow function`

```suggestion
  contractsListener = ({ contractName }: ISubscriber) => {
```